### ISSUE TITLE
non_blocking is ignored in GPU -> CPU copy (to pinned memory)

### ISSUE THREAD

**Issue by colesbury** [pytorch/pytorch#19625]
## 🐛 Bug

PyTorch master (d8729ef) Apr 23, 2019. (Also present in 1.0.0 and 1.0.1)

The copy from `x` to `y` should be non-blocking. It's not because `_s_copy_from_cuda` drops the `non_blocking` argument.

```python
import torch

x = torch.randn(10, device='cuda')
y = torch.empty(10, pin_memory=True)
y.copy_(x, non_blocking=True)
```

https://github.com/pytorch/pytorch/blob/e9c8f372c49431907ac525a2abbfe212e549f61e/aten/src/ATen/native/cuda/Copy.cu#L298-L305

**Comment by colesbury** [comment]
@li-roy

**Comment by smileeok** [comment]
Any update on this? Any other way of asynchronously copying GPU memory to CPU memory?

**Comment by li-roy** [comment]
@smileeok Sorry for the delay, fix will be out soon.

**Comment by li-roy** [comment]
closed by #20305
